### PR TITLE
chiaki: 1.3.0 -> 2.0.1 & touchpad support

### PR DESCRIPTION
**Pull request by Emantor**
###### Motivation for this change
Version bump & DualShock4 touchpad support under Linux.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/chiaki/default.nix [hunk 1]
```diff
@@ -1,23 +1,26 @@
-{ lib, mkDerivation, fetchFromGitHub
-, cmake, ffmpeg, libopus, qtbase, qtmultimedia, qtsvg, pkgconfig, protobuf
-, python3Packages, SDL2 }:
+{ lib, mkDerivation, fetchgit
+, cmake, ffmpeg, libevdev ? null, libopus, udev ? null, qtbase, qtmultimedia
```
Comment (SuperSandro2000) [inline pkgs/games/chiaki/default.nix:2]:
```suggestion
, cmake, ffmpeg, libevdev, libopus, udev, qtbase, qtmultimedia
```

File: pkgs/games/chiaki/default.nix [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+, qtsvg , pkgconfig, protobuf , python3Packages, SDL2, stdenv }:
```
Comment (SuperSandro2000) [inline pkgs/games/chiaki/default.nix:3]:
```suggestion
, qtsvg , pkg-config, protobuf , python3Packages, SDL2, stdenv }:
```

File: pkgs/games/chiaki/default.nix [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 
+assert stdenv.hostPlatform.isLinux -> (udev != null && libevdev != null);
+
```
Comment (SuperSandro2000) [inline pkgs/games/chiaki/default.nix:6]:
```suggestion
```

File: pkgs/games/chiaki/default.nix [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
     rev = "v${version}";
-    owner = "thestr4ng3r";
-    repo = "chiaki";
     fetchSubmodules = true;
-    sha256 = "07w7srxxr8zjp91p5n1sqf4j8lljfrm78lz1m15s2nzlm579015h";
+    sha256 = "0l532i9j6wmzbxqx7fg69kgfd1zy1r1wlw6f756vpxpgswivi892";
   };
 
   nativeBuildInputs = [
     cmake pkgconfig protobuf python3Packages.python python3Packages.protobuf
```
Comment (SuperSandro2000) [inline pkgs/games/chiaki/default.nix:20]:
```suggestion
    cmake pkg-config protobuf python3Packages.python python3Packages.protobuf
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -25609,7 +25609,10 @@ in
 
   chessx = libsForQt5.callPackage ../games/chessx { };
 
-  chiaki = libsForQt5.callPackage ../games/chiaki { };
+  chiaki = libsForQt5.callPackage ../games/chiaki {
+    libevdev = if stdenv.isDarwin then null else libevdev;
+    udev = if stdenv.isDarwin then null else udev;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25615]:
```suggestion
  chiaki = libsForQt5.callPackage ../games/chiaki { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107983` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chiaki</li>
  </ul>
</details>

```
CMake Error at /nix/store/05nmn6v1hrk56zqs61s2awvykha51fx9-qtbase-5.12.7-dev/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package):
  By not providing "FindQt5MacExtras.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Qt5MacExtras", but CMake did not find one.

  Could not find a package configuration file provided by "Qt5MacExtras" with
  any of the following names:

    Qt5MacExtrasConfig.cmake
    qt5macextras-config.cmake

  Add the installation prefix of "Qt5MacExtras" to CMAKE_PREFIX_PATH or set
  "Qt5MacExtras_DIR" to a directory containing one of the above files.  If
  "Qt5MacExtras" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  gui/CMakeLists.txt:7 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/nix-build-chiaki-2.0.1.drv-0/chiaki/build/CMakeFiles/CMakeOutput.log".
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chiaki</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (Emantor) [reply pkgs/top-level/all-packages.nix:25615]: won't this break darwin since `evdev` is only available on Linux?

**Comment by prusnak** [comment]
>   By not providing "FindQt5MacExtras.cmake" in CMAKE_MODULE_PATH this project

Add `qtmacextras` to `buildInputs` for darwin

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (prusnak) [reply pkgs/top-level/all-packages.nix:25615]: Nix has lazy evaluation, so unless you call the symbol it is not required.

File: pkgs/games/chiaki/default.nix [hunk 6]
```diff
@@ -1,23 +1,27 @@
...
   ];
-  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ];
+  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ]
+                ++ optionals stdenv.hostPlatform.isLinux [ libevdev udev]
+                ++ optional (stdenv.isDarwin) qtmacextras;
```
Comment (prusnak) [inline pkgs/games/chiaki/default.nix:24]: You need to add `qtmacextras` at the top of the file
+
small consistency suggestion:

```suggestion
                ++ optionals stdenv.isLinux [ libevdev udev ]
                ++ optionals stdenv.isDarwin [ qtmacextras ];
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/games/chiaki/default.nix (hunk 6 above)
Comment (Emantor) [reply pkgs/games/chiaki/default.nix]: done.

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (Emantor) [reply pkgs/top-level/all-packages.nix]: I see, other package using this scheme do this because they need to modify the configure stage for this.

**Review by prusnak — APPROVED** [review]

File: pkgs/games/chiaki/default.nix [hunk 7]
```diff
@@ -1,23 +1,25 @@
...
   };
 
   nativeBuildInputs = [
-    cmake pkgconfig protobuf python3Packages.python python3Packages.protobuf
+    cmake pkg-config protobuf python3Packages.python python3Packages.protobuf
   ];
-  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ];
+  buildInputs = [ ffmpeg libopus qtbase qtmultimedia qtsvg protobuf SDL2 ]
+                ++ optionals stdenv.hostPlatform.isLinux [ libevdev udev]
+                ++ optionals (stdenv.isDarwin) [ qtmacextras ];
```
Comment (SuperSandro2000) [inline pkgs/games/chiaki/default.nix:22]:
```suggestion
    ++ optionals stdenv.hostPlatform.isLinux [ libevdev udev]
    ++ optionals (stdenv.isDarwin) [ qtmacextras ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107983` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>chiaki</li>
  </ul>
</details>

```
applying patch /nix/store/d8nj29wc99kwsj5qkhdihr956wkfd3jy-0001-qtbase-mkspecs-mac.patch
patching file mkspecs/common/mac.conf
patching file mkspecs/features/mac/default_post.prf
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch. 
1 out of 1 hunk ignored -- saving rejects to file mkspecs/features/mac/default_post.prf.rej
patching file mkspecs/features/mac/default_pre.prf
patching file mkspecs/features/mac/sdk.mk
patching file mkspecs/features/mac/sdk.prf
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>chiaki</li>
</ul>
</details>

**Comment by prusnak** [comment]
> Result of `nixpkgs-review pr 107983` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

darwin build is broken because patches do not apply after mistake in https://github.com/NixOS/nixpkgs/pull/107483 , this the build failure is not related to this PR

**Pull request merged**